packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -387,6 +411,20 @@ public void onConnectionSuspended(int cause) {
 
   @Override
   public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
+    if (requestCode == REQUEST_CODE_RESOLVE_ERROR) {
+      // Deal with result of `onConnectionFailed` error resolution.
+      resolvingError = false;
+      if (resultCode == Activity.RESULT_OK) {
+        // Make sure the app is not already connected or attempting to connect
+        if (!googleApiClient.isConnecting() && !googleApiClient.isConnected()) {
+          googleApiClient.connect();
+        }
+      } else if (pendingOperation != null && pendingOperation.method.equals(METHOD_INIT)) {
```

Comment:
In this case, should we `finishWithError()` no matter what the state of `pendingOperation`?